src/cryptoadvance/specter/wallet.py

```diff
@@ -0,0 +1,510 @@
+        try:
+            self.cli.lockunspent(True, self.pending_psbts[txid]["tx"]["vin"])
+        except:
+            # UTXO was spent
+            pass
+        del self.pending_psbts[txid]
+        self.save_to_file()
+
+    def update_pending_psbt(self, psbt, txid, raw, device_name):
+        if txid in self.pending_psbts:
+            self.pending_psbts[txid]["sigs_count"] += 1
+            self.pending_psbts[txid]["base64"] = psbt
+            if device_name:
+                if "devices_signed" not in self.pending_psbts[txid]:
+                    self.pending_psbts[txid]["devices_signed"] = []
+                self.pending_psbts[txid]["devices_signed"].append(device_name)
+            if "hex" in raw:
+                self.pending_psbts[txid]["raw"] = raw["hex"]
+            self.save_to_file()
+
+    def save_pending_psbt(self, psbt):
+        self.pending_psbts[psbt["tx"]["txid"]] = psbt
+        self.cli.lockunspent(False, psbt["tx"]["vin"])
+        self.save_to_file()
+
+    @property
+    def txlist(self):
+        ''' The last 1000 transactions for that wallet - filtering out change addresses transactions and duplicated transactions (except for self-transfers)
+            This list is used for the wallet `txs` tab to list the wallet transacions.
+        '''
+        txidlist = []
+        txlist = []
+        for tx in self.transactions:
+            if tx["is_change"] == False and (tx["is_self"] or tx["txid"] not in txidlist):
+                txidlist.append(tx["txid"])
+                txlist.append(tx)
+        return txlist
+
+    @property
+    def rescan_progress(self):
+        """Returns None if rescanblockchain is not launched,
+           value between 0 and 1 otherwise
+        """
+        if self.info is {} or "scanning" not in self.info or self.info["scanning"] == False:
+            return None
+        else:
+            return self.info["scanning"]["progress"]
+
+    def getnewaddress(self, change=False):
+        label = "Change" if change else "Address"
+        if change:
+            self.change_index += 1
+            index = self.change_index
+        else:
+            self.address_index += 1
+            index = self.address_index
+        address = self.get_address(index, change=change)
+        self.setlabel(address, "{} #{}".format(label, index))
+        if change:
+            self.change_address = address
+        else:
+            self.address = address
+        self.save_to_file()
+        return address
+
+    def get_address(self, index, change=False):
+        pool = self.change_keypool if change else self.keypool
+        if pool < index + WALLET_CHUNK:
+            self.keypoolrefill(pool, index + WALLET_CHUNK, change=change)
+        desc = self.change_descriptor if change else self.recv_descriptor
+        if self.is_multisig:
+            desc = sort_descriptor(self.cli, desc, index=index, change=change)
+            return self.cli.deriveaddresses(desc)[0]
```

Comment:
`sort_descriptor` doesn't work correctly if you provide "sortedmulti" there.
In v20 it causes invalid ordering of pubkeys, in 19 - rpc error.
And sorting descriptor is not necessary for v20 so I suggest the following:
```py
    if self.is_multisig:
            # first try with sortedmulti
            try:
                addr = self.cli.deriveaddresses(desc)[0]
            except:
                # if sortedmulti is not supported
                desc = desc.replace("sortedmulti", "multi")
                desc = sort_descriptor(self.cli, desc, index=index, change=change)
                addr = self.cli.deriveaddresses(desc)[0]
            return addr
```